Build of examples is broken

**Version of neli**
Tested:
`master` branch
`neli-v0.6.1` branch

**Describe the bug**

```
   Compiling neli v0.6.1 (/home/fmp/opensrc/rust/linux/neli)
error[E0432]: unresolved import `neli::socket::tokio`
  --> examples/async_nl80211.rs:15:13
   |
15 |     socket::tokio::NlSocket,
   |             ^^^^^ could not find `tokio` in `socket`

For more information about this error, try `rustc --explain E0432`.
error: could not compile `neli` due to previous error
warning: build failed, waiting for other jobs to finish...
error: build failed
```

**To Reproduce**

`$ cargo build --examples`

or 

`$ cargo test`

**Expected behavior**

Build & run unit tests without error. :-)

**Additional context**

Running on Linux Mint 19 (Ubuntu 18.04)

--- a/.githooks/pre-commit
+++ b/.githooks/pre-commit
@@ -4,6 +4,8 @@ cargo fmt -- --check \
     && cargo clippy --all-targets --all-features -- -D warnings \
     && cargo build \
     && cargo build --all-targets --all-features \
+    && cargo build --examples \
+    && cargo build --examples --features=async \
     && cargo test --all-targets --all-features \
     || exit 1
 
@@ -15,8 +17,9 @@ cargo fmt -- --check \
     && ./target/debug/route-list \
     && ./target/debug/ctrl-list \
     && ./target/debug/error_packet \
+    && ./target/debug/nl80211 \
     && cargo build --all-targets --all-features \
-    && ./target/debug/async_nl80211) \
+    && ./target/debug/nl80211) \
     || exit 1
 
 (cd neli-proc-macros \

--- a/.github/workflows/main.yml
+++ b/.github/workflows/main.yml
@@ -38,6 +38,8 @@ jobs:
           - cargo build
           - cargo build --features=async
           - cargo build --all-features
+          - cargo build --examples
+          - cargo build --examples --all-features
           - RUST_TEST_THREADS=1 cargo test --all-features
           - (cd neli-proc-macros && cargo fmt -- --check)
           - (cd neli-proc-macros && cargo build)
@@ -86,6 +88,8 @@ jobs:
           - cargo build
           - cargo build --features=async
           - cargo build --all-features
+          - cargo build --examples
+          - cargo build --examples --all-features
           - RUST_TEST_THREADS=1 cargo test --all-features
         toolchain:
           - nightly

--- a/examples/Cargo.toml
+++ b/examples/Cargo.toml
@@ -37,9 +37,8 @@ name = "error_packet"
 path = "error_packet.rs"
 
 [[bin]]
-name = "async_nl80211"
-path = "async_nl80211.rs"
-required-features = ["async"]
+name = "nl80211"
+path = "nl80211.rs"
 
 [dependencies.tokio]
 version = "1"

--- a/examples/nl80211.rs
+++ b/examples/nl80211.rs
@@ -5,6 +5,8 @@ use log::Level;
 #[cfg(feature = "logging")]
 use simple_logger::init_with_level;
 
+#[cfg(feature = "async")]
+use neli::socket::tokio::NlSocket;
 use neli::{
     consts::{
         nl::{GenlId, NlmF, NlmFFlags},
@@ -12,7 +14,6 @@ use neli::{
     },
     genl::Genlmsghdr,
     nl::{NlPayload, Nlmsghdr},
-    socket::tokio::NlSocket,
     socket::NlSocketHandle,
     types::GenlBuffer,
 };
@@ -38,6 +39,7 @@ fn handle(msg: Nlmsghdr<GenlId, Genlmsghdr<Nl80211Command, Nl80211Attribute>>) {
     println!("msg={:?}", msg.nl_type);
 }
 
+#[cfg(feature = "async")]
 #[tokio::main]
 async fn main() -> Result<(), Box<dyn Error>> {
     #[cfg(feature = "logging")]
@@ -73,6 +75,51 @@ async fn main() -> Result<(), Box<dyn Error>> {
     let msgs = ss.recv(&mut buffer).await?;
     println!("msgs: {:?}", msgs);
     for msg in msgs {
+        if let NlPayload::Err(e) = msg.nl_payload {
+            if e.error == -2 {
+                println!(
+                    "This test is not supported on this machine as it requires nl80211; skipping"
+                );
+            } else {
+                return Err(Box::new(e) as Box<dyn Error>);
+            }
+        } else {
+            handle(msg);
+        }
+    }
+    Ok(())
+}
+
+#[cfg(not(feature = "async"))]
+fn main() -> Result<(), Box<dyn Error>> {
+    #[cfg(feature = "logging")]
+    init_with_level(Level::Trace)?;
+
+    let mut sock = NlSocketHandle::connect(
+        NlFamily::Generic, /* family */
+        Some(0),           /* pid */
+        &[],               /* groups */
+    )?;
+    let family_id = sock.resolve_genl_family("nl80211")?;
+
+    let req = Nlmsghdr::new(
+        /* len */ None,
+        /* type */ family_id,
+        /* flags */ NlmFFlags::new(&[NlmF::Request, NlmF::Dump, NlmF::Ack]),
+        /* seq */ Some(1),
+        /* pid */ Some(0),
+        /* payload */
+        NlPayload::Payload(Genlmsghdr::<Nl80211Command, Nl80211Attribute>::new(
+            /* cmd */ Nl80211Command::GetWiPhy,
+            /* version */ 1,
+            /* attrs */ GenlBuffer::new(),
+        )),
+    );
+
+    sock.send(req)?;
+
+    for msg in sock.iter(false) {
+        let msg = msg?;
         if let NlPayload::Err(e) = msg.nl_payload {
             if e.error == -2 {
                 println!(

